token.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"io/ioutil"
+	"strings"
+)
+
+func NewNoOpTokenProvider() *StaticToken {
+	return &StaticToken{token: ""}
+}
+
+type TokenProvider interface {
+	Get() (string, error)
+}
```

Comment:
Interfaces should be defined where they are consumed.
As far as I can see that's in the main.go :)

Reply:
agreed, fixed